app/services/translation.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class Translation
+  def self.price_category(price)
+    price = price.at_xpath("category").try(:text)
```

Comment:
Bei der separierten Klasse hätte ich eine Abkapselung vom Import erwartet.
Finde, dass der `price ` hier direkt der Parameter sein sollte und das `price.at_xpath("category").try(:text)` noch vorher stattfinden.
Nur so bleibt die Klasse in sich abgeschlossen und wir wiederverwendbar.

Und dann vermutlich `return '' if price.blank?`.

Reply:
unten wird auf ein anderes Attribute von :price Zugegriffen (:categorytext) daher reicht es leider nicht nur den price text reinzugeben in die Methode.

Ich glaube ich würde die Ursprüngliche Weiche in :poi_record belassen und in Translate auch nur den Text übergeben der zu Übersetzen ist.

Reply:
Die Klasse ist als Sammelklasse für alle möglichen Translation methods gedacht. Die je spezifisch auf inputfelder der tmb getrimmt sind. Wiederverwertbarkeit finde ich in diesem fall nicht so wichtig.
@marcometz Habe beim programmieren auch darüber nachgedacht die weiche für categorytext in poi_record zu belassen, mich aber letztendlich dagegen entschieden. Ich finde es besser alles an einem Platz zu haben. So wird es übersichtlicher und man sieht auf einen Blick was in welchem Fall passiert. Eine weitere Idee wäre es categorytext schon oben als Argument reinzugeben, das bedeutet die Methode mit zwei Argumenten aufzurufen. Wenn Du also drauf bestehst, werde ich categorytext wieder in poi_record zurück geben, finde es aber nicht so gut.

Reply:
Aber der import sollte natürlich trotzdem abgekapselt werden. Habe ich in meinen letzten commits auch gemacht